packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -32,7 +32,7 @@ import {LoadContext} from '@docusaurus/types';
 import {validateBlogPostFrontMatter} from './blogFrontMatter';
 
 export function truncate(fileString: string, truncateMarker: RegExp): string {
-  return fileString.split(truncateMarker, 1).shift()!;
+  return fileString.split(truncateMarker, 1).shift() as string;
```

Comment:
I don't like this change much. This basically do the exact same thing, so if "as string" does not emit a warning, we'd rather allow usage of !

Reply:
Should I add `eslint-ignore` then...?

Reply:
you can disable the no-non-null rule totally. If we allow "as", we should allow "!", so in any case we'd rather be consistent